src/NuGet.Core/NuGet.Protocol/README.md

```diff
@@ -1 +1,190 @@
-NuGet's implementation for interacting with feeds. Contains functionality for all feed types.
+# NuGet.Protocol
+
+NuGet.Protocol is a NuGet client SDK library that provides a set of APIs for interacting with NuGet feeds. It provides a way for developers to query NuGet feeds to discover packages and their dependencies, and also to download packages and their associated assets.
+
+## Getting started
+
+NuGet.Protocol can be installed from the NuGet Package Manager or using the dotnet CLI:
+
+```
+dotnet add package NuGet.Protocol
+```
+
+## Usage
+
+At the center of this library are the PackageSource and SourceRepository types, which represent a NuGet source that may be a file source or an http based source implementing the V2 or [V3](https://learn.microsoft.com/en-us/nuget/api/overview#versioning) protocol.
+
+```
+PackageSource localSource = new PackageSource("D:\LocalSource");
+SourceRepository localRepository = Repository.Factory.GetCoreV3(localSource);
+
+SourceRepository repository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
+```
+
+The SourceRepository then has a GetResourceAsync method that you can use to acquire implementations of INuGetResource that often correlate to resources on V3.
```

Comment:
I'd prefer simpler language to be friendly to folks that aren't familiar with English:

```suggestion
The SourceRepository then has a GetResourceAsync method that you can use to acquire implementations of INuGetResource that often are V3 resources.
```